src/app/derive/derive.component.ts

```diff
@@ -38,6 +40,13 @@ export class DeriveComponent implements OnInit {
 
     // Set derive to true
     this.globalVars.derive = true;
+
+    this.route.queryParams.subscribe((params) => {
+      if (params.derivedPublicKeyBase58Check) {
+        this.derivedPublicKeyBase58Check = params.derivedPublicKeyBase58Check;
```

Comment:
do we need the local `derivedPublicKeyBase58Check` if we also store it in globalVars?